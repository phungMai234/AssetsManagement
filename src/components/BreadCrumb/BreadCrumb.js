import React from 'react';
import { useCurrentRoute } from 'react-navi';
import BreadcrumbWrapper from './BreadCrumb.style';

const BreadCrumb = () => {
  const {
    data: { breadcrumb },
  } = useCurrentRoute();
  if (!breadcrumb) {
    return null;
  }
  return (
    <BreadcrumbWrapper>
      <ul className="breadcrumb">
        {breadcrumb.map((link, index) => {
          const isLastItem = index === breadcrumb.length - 1;
          const href = link.url && !isLastItem ? { href: link.url } : {};
          return (
            <li
              className="breadcrumb-item"
              aria-current="page"
              key={link.url + index}
            >
              <a {...href} className={`${isLastItem ? 'active' : ''}`}>
                {link.title}
              </a>
            </li>
          );
        })}
      </ul>
    </BreadcrumbWrapper>
  );
};
export default BreadCrumb;
