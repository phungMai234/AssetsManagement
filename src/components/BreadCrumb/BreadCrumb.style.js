import styled from 'styled-components';

export default styled.nav`
  .breadcrumb {
    background: #fff;
    padding: 0 !important;
    margin: 0;
    margin-top: 0px;

    .breadcrumb-item {
      a {
        color: #707070;
        font-weight: 300;
        text-decoration: none;

        &.active {
          color: #007bff;
        }

        &:hover {
          text-decoration: underline;
        }
      }
    }

    .breadcrumb-item + .breadcrumb-item::before {
      content: '>';
    }
  }
  h3 {
    font-weight: 300 !important;
  }
`;
