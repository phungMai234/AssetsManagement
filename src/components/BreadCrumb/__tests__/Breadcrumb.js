import React from 'react';

import { render } from '@testing-library/react';

import BreadCrumb from '../BreadCrumb';

test('should render correctly', () => {
  const container = render(
    <BreadCrumb
      breadcrumb={[
        {
          url: '/dashboard/delivery_reports',
          title: 'Danh sách biên bản',
        },
        {
          url: '',
          title: 'Tạo mới',
        },
      ]}
    />,
  );
  expect(container.container).toMatchSnapshot();
});
