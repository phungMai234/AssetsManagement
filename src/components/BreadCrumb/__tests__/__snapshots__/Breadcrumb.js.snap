// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div>
  <nav
    class="sc-bdfBwQ pRJKX"
  >
    <ul
      class="breadcrumb"
    >
      <li
        aria-current="page"
        class="breadcrumb-item"
      >
        <a
          class=""
          href="/dashboard/delivery_reports"
        >
          Danh sách biên bản
        </a>
      </li>
      <li
        aria-current="page"
        class="breadcrumb-item"
      >
        <a
          class="active-tab"
        >
          Tạo mới
        </a>
      </li>
    </ul>
  </nav>
</div>
`;
